superset-frontend/src/SqlLab/actions/sqlLab.js

```diff
@@ -1033,6 +1033,19 @@ export function queryEditorSetSql(queryEditor, sql) {
   return { type: QUERY_EDITOR_SET_SQL, queryEditor, sql };
 }
 
+export function formatQuery(queryEditor) {
+  return function (dispatch, getState) {
+    const { sql } = getUpToDateQuery(getState(), queryEditor);
+    return SupersetClient.post({
```

Comment:
why couldn't we just us a npm package to do this work vs. creating a whole new endpoint?

Reply:
> why couldn't we just us a npm package to do this work vs. creating a whole new endpoint?

@hughhhh We chose server-side formatting over client-side formatting for several reasons:

1. It reduces the need to maintain additional NPM dependencies.
2. It reduces the package size.
3. It ensures that SQL formatting is consistent across surfaces in "view query", using sqlparse.

If you and other community members believe that an NPM package solution is better, I am happy to make changes.

Reply:
There seems to be a number of pros with having this on the backend including: custom formatting, dialect specific formatting, and consistency throughout the application.

@villebro has experience working with frontend SQL formatting and ran into a number of issues.